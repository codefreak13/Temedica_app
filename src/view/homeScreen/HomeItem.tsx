import React, { FC } from 'react';
import styled from 'styled-components/native'
import moment from 'moment'
import { colors } from './../../colors'

export const Text = styled.Text`
color: ${colors.BLACK};
`
export const View = styled.View``

const TinyText = styled(Text)`
font-size: 12px
`
const BoldText = styled(Text)`
font-weight: bold
`
const Wrapper = styled(View)`
margin-vertical: 10px;
borderWidth: 1px;
borderColor: ${colors.GREY};
borderRadius: 3px;
padding:10px
`
const MainWrapper = styled(View)`
flex-direction: row;
justify-content: space-between
`
const ItemWrapper = styled(View)`
width: 80%
`
const DescriptionView = styled(View)`
background-color: ${colors.DARK_GREY};
padding: 10px;
margin-top: 10px;
`

export interface IProps {
    id: string,
    name: string,
    diseases: Array<string>,
    description: string,
    released: string
}

const ListItem: FC<IProps> = ({ name, diseases, description, released }) => {
    const formatDate = moment(released).format("DD/MM/YYYY")
    return (
        <Wrapper>
            <MainWrapper>
                <ItemWrapper>
                    <BoldText>{name}</BoldText>
                    <TinyText>
                        {diseases && diseases.join(', ')}
                    </TinyText>
                </ItemWrapper>
                <BoldText>{formatDate}</BoldText>
            </MainWrapper>
            <DescriptionView>
                <Text>{description}</Text>
            </DescriptionView>
        </Wrapper>
    )
};


export default ListItem;
