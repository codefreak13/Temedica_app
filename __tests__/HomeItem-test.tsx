import React from 'react';
import { render } from '@testing-library/react-native';
import HomeItem from '../src/view/homeScreen/HomeItem';

it('HomeItem component renders correctly', () => {
    const snap = render(
        <HomeItem  name={'Panadol'} diseases={["malaria, 'fever"]} description={'relieves headache'} released={'10/10/1995'} />
    ).toJSON();
    expect(snap).toMatchSnapshot();
});