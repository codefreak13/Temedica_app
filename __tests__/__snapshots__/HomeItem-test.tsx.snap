// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeItem component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#B4B4B4",
        "borderRadius": 3,
        "borderWidth": 1,
        "marginVertical": 10,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": "80%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontWeight": "bold",
            },
          ]
        }
      >
        Panadol
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 12,
            },
          ]
        }
      >
        malaria, 'fever
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontWeight": "bold",
          },
        ]
      }
    >
      10/10/1995
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F1F1F1",
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
        ]
      }
    >
      relieves headache
    </Text>
  </View>
</View>
`;
